system/lib/pthread/library_pthread.c

```diff
@@ -743,15 +667,70 @@ em_queued_call* emscripten_async_waitable_run_in_main_runtime_thread_(
   return q;
 }
 
-typedef struct DispatchToThreadArgs {
-  pthread_t target_thread;
-  em_queued_call* q;
-} DispatchToThreadArgs;
+int emscripten_dispatch_to_thread_ptr(pthread_t target_thread,
+                                      void (*func)(void*),
+                                      void* arg) {
+  // Can't be a null pointer here, and can't be
```

Comment:
If you implement this function above where the old _emscripten_do_dispatch_to_thread does it allow github to show this diff more reasonably?

Reply:
Would be good to retain the location so that this can be diffed easier, not completely sure what happens here